src/dnx/pal.linux.cpp

```diff
@@ -5,111 +5,32 @@
 #include <assert.h>
 #include <dlfcn.h>
 
-void GetNativeBootstrapperDirectory(LPTSTR szPath)
+LPTSTR GetNativeBootstrapperDirectory()
 {
-    ssize_t ret = readlink("/proc/self/exe", szPath, PATH_MAX - 1);
+    LPTSTR szPath = (LPTSTR)calloc(PATH_MAX + 1, sizeof(TCHAR));
+    ssize_t ret = readlink("/proc/self/exe", szPath, PATH_MAX);
 
     assert(ret != -1);
 
-    szPath[ret] = '\0';
+    // readlink does not null terminate the path
+    szPath[ret] = _T('\0');
 
     size_t lastSlash = 0;
 
-    for (size_t i = 0; szPath[i] != '\0'; i++)
+    for (size_t i = 0; i < ret && szPath[i] != _T('\0'); i++)
```

Comment:
why are we looking for the last slash from the first character and not from the last?